### PR TITLE
Feature/Add TheBanner component

### PR DESCRIPTION
**Pull request by mateuszRybczonek**
This PR adds a single Banner component that can be reused within the apps.
For how the props work check Storybook.

Error:
Screenshot 2022-07-20 at 12 29 00

Warning:
Screenshot 2022-07-20 at 12 29 14

Success:
Screenshot 2022-07-20 at 12 29 25

Real example:
Screenshot 2022-07-20 at 12 29 42

## Usage:
1. Mount the component at the root of your App and register listeners from the notifications util.
```
// App.vue

<template>
  <TheBanner ref="theBanner" />
  <router-view />
</template>

<script setup>
import TheBanner from '@uicommon/components/TheBanner'
import { addObserver, ENotification, removeObserver } from '@uicommon/utils/notifications'
import { ref, onMounted, onUnmounted } from 'vue'

const theBanner = ref()

function showBanner(object) {
  theBanner.value?.show(object)
}

function hideBanner() {
  theBanner.value?.hide()
}

onMounted(() => {
  addObserver(ENotification.showBanner, showBanner)
  addObserver(ENotification.hideBanner, hideBanner)
})

onUnmounted(() => {
  removeObserver(ENotification.showBanner, showBanner)
  removeObserver(ENotification.hideBanner, hideBanner)
})
</script>
```

2. Call `showBanner` from anywhere within the app
```
async function login() {
    try {
      const isAuthenticated = await createKeypairAndCheckAuth(
        email.value.toLowerCase(),
        password.value
      )

      if (isAuthenticated) {
        localStorage.setItem('isAuthed', 'true')
      } else {
        postNotification(ENotification.showBanner, { message: t('$.errors.login_failed') })
      }
    } catch (e) {
      postNotification(ENotification.showBanner, { message: t('$.errors.login_failed') })
    }
  }
}
```

you can also pass a custom component to the banner
```
import KycBanner from 'components/banners/KycBanner.vue'

postNotification(ENotification.showBanner, { contentComponent: KycBanner })
```

By using this component as shown above we can avoid mounting multiple banners, we can use it from anywhere, we don't have to keep any extra props in the store.

### CODE REVIEW

**Review by zeeshan595 — APPROVED** [review]

**Review by Paterick — APPROVED** [review]

File: src/components/TheBanner.vue [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<template>
+	<Transition>
```
Comment (robbiecarlton) [inline src/components/TheBanner.vue:2]: looks like tabs here, can you convert to spaces

File: src/components/icons/ExIcon.vue [hunk 2]
```diff
@@ -1,33 +1,27 @@
 <template>
-  <Icon fill="#606C8B" :style="style" viewBox="2 2 8 8">
-    <path d="M6 5.3l2.5-2.5.7.7L6.7 6l2.5 2.5-.7.7L6 6.7 3.5 9.2l-.7-.7L5.3 6 2.8 3.5l.7-.7L6 5.3z" />
-  </Icon>
+	<Icon :fill="color" :style="size" viewBox="2 2 8 8">
```
Comment (robbiecarlton) [inline src/components/icons/ExIcon.vue:2]: tabs

**Review by robbiecarlton — APPROVED** [review]
The code looks good, and banner looks good.

I'm not sure this one belongs in the common library though. If it's an error banner for the host console login page specifically, the other apps don't have login pages as part of their code base. Maybe I'm missing something, and this has a more general purpose. But I'm not seeing this in the designs for this release.

I'm going to be out for the rest of the week, so don't want to block this. If @celestialhanley has signed off on this being a component we're using across all apps, then I'm happy for it to be merged. If not, then I'd say let's leave it in host console til we know we need it elsewhere

**Comment by mateuszRybczonek** [comment]
@robbiecarlton let's talk about tabs vs. spaces in the next UI/UX Guild meeting, I really don't understand why would you ever use spaces, but maybe I am missing something. I will leave tabs for now, but I am open to discuss and change it everywhere later when I understand why.

**Pull request merged**